corgi/tasks/pnc.py

```diff
@@ -123,3 +124,62 @@ def slow_fetch_pnc_sbom(purl: str, product_data, build_data, sbom_data) -> None:
 
     # Save component taxonomy
     root_component.save_component_taxonomy()
+
+
+@app.task(base=Singleton, autoretry_for=RETRYABLE_ERRORS, retry_kwargs=RETRY_KWARGS)
+def slow_handle_pnc_errata_released(erratum_id: int, erratum_status: str) -> None:
+    # Check that the erratum is released
+    if erratum_status != "SHIPPED_LIVE":
+        raise ValueError(f"Invalid status {erratum_status} for erratum {erratum_id}")
+
+    # Get the purl(s) of the related component(s) from the erratum's Notes field
+    et = ErrataTool()
+    notes = et.get_erratum_notes(erratum_id)
+
+    related_purls = set()
+    for ref in notes.get("manifest", {}).get("refs", {}):
+        if ref["type"] == "purl":
+            related_purls.add(ref["uri"])
+
+    if not related_purls:
+        raise ValueError(f"Erratum {erratum_id} had no associated purls")
+
+    # Check that there's a component matching the purls
+    root_components: set[Component] = set()
+    for purl in related_purls:
+        # There should only be one root component of SOURCE type
+        components = Component.objects.filter(meta_attr__purl_declared=purl).filter(
+            type=Component.Type.MAVEN
+        )
+        if components.count() == 0:
+            logger.warning(
+                f"Erratum {erratum_id} refers to purl {purl} which matches no components"
+            )
+            continue
+        if components.count() > 1:
+            logger.warning(
+                f"Erratum {erratum_id} refers to purl {purl} which matches {components.count()}"
+                " components; only handling the first"
+            )
+        root_components.add(components[0])
+
+    # Update relations with this erratum
+    # TODO: Is just the root component okay here, or do I need to recreate the entire tree?
+    for component in root_components:
+        if component.software_build is None:
+            logger.warning(
+                f"Component {component.purl} has no build, can't relate erratum {erratum_id}"
+            )
```

Comment:
I think it's better to raise an error here. Alternatively, you can do:
```suggestion
            logger.warning(
                f"Component {component.purl} has no build, can't relate erratum {erratum_id}"
            )
            continue
```
And then the code below doesn't need to be wrapped in an else block or indented.